### ISSUE TITLE
Card suggestion based on location

### ISSUE THREAD

**Issue by TheLastProject** [TheLastProject/Catima#456]
It would be interesting to store a combination of:

1. Timestamp
2. GPS coordinates
3. Card ID

And whenever a card is opened, save a new entry to this.

Then, we could every 5 minutes or so check the current GPS coordinates and suggest a card that matches nearby coordinates.

For example:
You go to the grocery store, and open your grocery store card. Catima stores the GPS location where you opened the card. So, the next time you go to the grocery store, Catima can create an Android notification telling you that it thinks you're at the grocery store and allowing you to open the grocery store card with a single tap.

This is more of an experiment and will need a lot of tweaking to figure out how to make it helpful but not creepy (even though we will of course not send any data online, we don't want to scare any user)

**Comment by djechelon** [comment]
Geofence API doesn't require you to check GPS every x minutes. Android does it for you.

Consider that unless you buy at the same shop(s), you may want to add the locations of all shops of the same brand in your region.

Consider that most users consider creepy to allow permanent GPS access, but being an optional feature it's not a problem to refuse.

My experience: I don't have problems with loyalty cards (the cash attendant often asks me if I own a card), but rather with discount coupons. I forgot to show some precious coupons often.

**Comment by TheLastProject** [comment]
Hmm... looking at https://developer.android.com/training/location/geofencing#re-register-geofences-only-when-required this may depend on GMS...

**Comment by TheLastProject** [comment]
Not forbidden, as long as no non-free libraries are added. And there is microG. But I would prefer to avoid any GMS dependency in any way whenever possible. Just something I will need to research deeper to know exactly what decision we'll be making if we use this.

**Comment by djechelon** [comment]
I fully get your point and I agree about how bad the requirement is to work only on Google-bloated devices.

But my view was about effort required vs the result (how many people reached by the feature, or using it).

Your point, if I interpret correctly, is that you don't like to offer a feature if that has limitations. All or none. Not even a warning "feature is _currently_ unavailable". That is a good point and the best way to offer privacy.

Then, we could think about getting the best of both worlds if it was ever possible

1. If user has Google suite, offer to use it
2. If the user doesn't want geolocation/geofencing, don't require to install extra apps
3. If the user doesn't have Google suite, offer to install a third party open source app
4. Don't break existing app functionality

Of course if it was possible to use geofencing with a 3rd party open source library, voilà, we are done 🧙‍♂️

But again, it is important to understand if it is legal to **link** an application to GMS and distribute it over F-Droid, even when providing an alternate to non-GMS users.

I don't know if you get my view over this.

**Comment by TheLastProject** [comment]
>But again, it is important to understand if it is legal to link an application to GMS and distribute it over F-Droid, even when providing an alternate to non-GMS users.

Well, that is the thing, it depends solely on the app source code. If it is a libre library, it's fine, if it requires a proprietary library to talk to GMS, that's not okay. Using GMS itself is not really something that changes acceptance on F-Droid (but REQUIRING it to work means you get the NonFreeDep antifeature, as GMS is proprietary).

**Comment by famewolf** [comment]
> Keep it simple and lightweight please. If I want bloatware I will use stocard and many other alternatives. I don`t need a location to see what cards I`m having.

Assuming you have 1 vote, I'll vote the opposite thus making both null and void.  While I prefer privacy I'd much rather have an open source app use my location all the time to show cards/coupons etc when I enter a store than  proprietary like stocard.    I find the feature useful.  You have the option to not enable the feature.

**Comment by Korb** [comment]
> How many Catima users are running a Google-free device? Is that audience worth additional effort?

This is a trick question. Because following **the majority** is fraught with hellish consequences. Most people prefer proprietary products. Most people do not care about the security of their personal data. Most of my fellow citizens do not know the laws that directly affect them and are not ready to fight for their rights. Most people do not create anything fundamentally new in this world. Most people do not have color vision defects, are able to perceive interfaces without the use of screen readers, do not use hearing aids, speech synthesis devices, and are able to move around without the help of a wheelchair.

There is no need to provide a _democratic_ argumentation of the majority for a non-commercial project. Because this way there will be a little more ruzzia in the world, where the majority doesn’t care what its government does.

---
Returning to the discussion of the functionality of the proposed function, I would like to draw attention to one point: the use of one map at different points of the **network company**. Thus, if you do not plan to select a category from OSM for each map with a dynamic filter when starting the application, you will have to build a certain heat map for each discount card, each time it is used, increasing the probability of its appearance in the list of those offered to the user (or users - in the case of synchronization of the card index). This looks quite complicated.

**Comment by famewolf** [comment]
Google Wallet now supports adding loyalty cards directly to it.   Tasker
can be setup to open google wallet when at certain gps coordinated or a
street address.   The issue has become moot.

**Comment by TheLastProject** [comment]
I disagree, "using Google Wallet and giving Google all your data" is definitely not an acceptable workaround :)

**Comment by famewolf** [comment]
My point was if Catima won't provide the feature then users will go to
something that does.  Heck perplexity.ai and others can even generate an
importable tasker profile needed to open the app when at the addresses.

**Comment by TheLastProject** [comment]
Contributions are welcome, I can't write everything on my own. 1 person with 0 budget only has so much time available :)
